core/metamodel.class.php

```diff
@@ -4219,40 +4219,75 @@ public static function AddMagicPlaceholders($aPlaceholders, $aExpectedArgs = nul
 			foreach ($aExpectedArgs as $expression)
 			{
 				$aName = explode('->', $expression->GetName());
 				if ($aName[0] == 'current_contact_id') {
 					$aPlaceholders['current_contact_id'] = UserRights::GetContactId();
-				}
-				if ($aName[0] == 'current_user') {
+				} else if ($aName[0] == 'current_user') {
 					array_push($aCurrentUser, $aName[1]);
-				}
-				if ($aName[0] == 'current_contact') {
+				} else if ($aName[0] == 'current_contact') {
 					array_push($aCurrentContact, $aName[1]);
 				}
 			}
 			if (count($aCurrentUser) > 0) {
-				$oUser = UserRights::GetUserObject();
-				$aPlaceholders['current_user->object()'] = $oUser;
-				foreach ($aCurrentUser as $sField) {
-					$aPlaceholders['current_user->'.$sField] = $oUser->Get($sField);
-				}
+				static::FillPlaceholders($aPlaceholders, 'current_user', UserRights::GetUserObject(), $aCurrentUser);
 			}
 			if (count($aCurrentContact) > 0) {
-				$oPerson = UserRights::GetContactObject();
-				$aPlaceholders['current_contact->object()'] = $oPerson;
-				foreach ($aCurrentContact as $sField) {
-					$aPlaceholders['current_contact->'.$sField] = $oPerson->Get($sField);
-				}
+				static::FillPlaceholders($aPlaceholders, 'current_contact', UserRights::GetContactObject(), $aCurrentContact);
 			}
 		}
 
 		return $aPlaceholders;
 	}
 
+	/**
+	 * @since 3.1 N°6824
+	 * @param array $aPlaceholders
+	 * @param string $sPlaceHolderPrefix
+	 * @param \DBObject|null $oObject
+	 * @param array $aCurrentUser
+	 *
+	 * @return void
+	 *
+	 */
+	private static function FillPlaceholders(array &$aPlaceholders, string $sPlaceHolderPrefix, $oObject, $aCurrentUser) {
+		$sPlaceHolderKey = $sPlaceHolderPrefix."->object()";
+		if (is_null($oObject)){
+			$aContext = [
+				"null object type" => $sPlaceHolderPrefix,
+				"fields" => $aCurrentUser,
+				"current_user_id" => UserRights::GetUserId()
+			];
+			IssueLog::Warning("Unresolved placeholders due to null object in current context", null,
+				$aContext);
+			$aPlaceholders[$sPlaceHolderKey] = \Dict::Format("PLACEHOLDER_CANNOT_BE_RESOLVED", $sPlaceHolderKey);
+			foreach ($aCurrentUser as $sField) {
+				$sPlaceHolderKey = $sPlaceHolderPrefix . "->$sField";
+				$aPlaceholders[$sPlaceHolderKey] = \Dict::Format("PLACEHOLDER_CANNOT_BE_RESOLVED", $sPlaceHolderKey);
+			}
+		} else {
+			$aPlaceholders[$sPlaceHolderKey] = $oObject;
+			foreach ($aCurrentUser as $sField) {
+				$sPlaceHolderKey = $sPlaceHolderPrefix . "->$sField";
+				if (MetaModel::IsValidAttCode(get_class($oObject), $sField)){
+					$aPlaceholders[$sPlaceHolderKey] = $oObject->Get($sField);
+				} else {
+					$aContext = [
+						"placeholder" => $sPlaceHolderKey,
+						"obj_class" => get_class($oObject),
+						"invalid_field" => $sField,
+					];
+					IssueLog::Warning("Unresolved placeholder due to invalid attribute", null,
+						$aContext);
+					$aPlaceholders[$sPlaceHolderKey] = \Dict::Format("PLACEHOLDER_CANNOT_BE_RESOLVED", $sPlaceHolderKey);
```

Comment:
```suggestion
				if (false === MetaModel::IsValidAttCode(get_class($oObject), $sField)){
					$aContext = [
						"placeholder" => $sPlaceHolderKey,
						"obj_class" => get_class($oObject),
						"invalid_field" => $sField,
					];
					IssueLog::Warning("Unresolved placeholder due to invalid attribute", null,
						$aContext);
					$aPlaceholders[$sPlaceHolderKey] = \Dict::Format("PLACEHOLDER_CANNOT_BE_RESOLVED", $sPlaceHolderKey);
					continue;
				}
				
				$aPlaceholders[$sPlaceHolderKey] = $oObject->Get($sField);	
```

We tend to prefer that kind of pattern (exiting/continuing condition first) to avoid indentation hell